dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java

```diff
@@ -449,6 +456,37 @@ public void run(DockstoreWebserviceConfiguration configuration, Environment envi
         // Initialize GitHub App Installation Access Token cache
         CacheConfigManager cacheConfigManager = CacheConfigManager.getInstance();
         cacheConfigManager.initCache();
+
+        if (configuration.getEsConfiguration().isIndexOnStart()) {
+            environment.lifecycle().addLifeCycleListener(new LifeCycle.Listener() {
+                @Override
+                public void lifeCycleStarted(LifeCycle event) {
+                    if (!ElasticSearchHelper.doMappingsExist()) {
+                        // A lock is used to prevent concurrent indexing requests in a deployment where multiple webservices start at the same time
+                        if (ElasticSearchHelper.acquireLock()) {
```

Comment:
Just for the sake of completeness, I should note that this code will allow all N tasks to potentially perform an ES index.   If they all pass `doMappingsExist` and then stall out for the right amounts of time before `acquireLock`, each could theoretically acquire the lock and index, one by one.

This situation will basically never happen in real life, and the penalty is minor (>1 nonoverlapping ES indexes), so no change necessary.